inference-engine/src/cldnn_engine/cldnn_config.cpp

```diff
@@ -221,6 +222,19 @@ void Config::UpdateFromMap(const std::map<std::string, std::string>& configMap)
             } else {
                 THROW_IE_EXCEPTION_WITH_STATUS(NotFound) << "Unsupported KEY_CLDNN_ENABLE_FP16_FOR_QUANTIZED_MODELS flag value: " << val;
             }
+        } else if (key.compare(CLDNNConfigParams::KEY_CLDNN_MAX_NUM_THREADS) == 0) {
+            int max_threads = std::thread::hardware_concurrency();
+            try {
+                int val_i = std::stoi(val);
+                if (val_i <= 0 || val_i > max_threads) {
+                    n_threads = max_threads;
+                } else {
+                    n_threads = val_i;
+                }
+            } catch (const std::exception&) {
+                THROW_IE_EXCEPTION << "Wrong value for property key " << CLDNNConfigParams::KEY_CLDNN_MAX_NUM_THREADS
+                                    << ". Expected only integer";
```

Comment:
please add a usage example here too: 
e.g., "Please specify the number of threads you want to use for build as integer. If you set this with a value out of range, it will be set as default number, i.e., maximum number of available threads."

Reply:
Are you saying we should set the number of threads to its default for the cases when invalid config value is given, such as string or character, and not throw an exception?

Reply:
no. I mean just add more info as message here to be shown to the user.